### DISCUSSION TITLE
It is not possible to add multiple examples of requests when using a Depends in the body

### DISCUSSION THREAD

**Discussion by w11114m** [fastapi/fastapi#8430] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [x] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends, Request, Body
from pydantic import BaseModel, Field
import uvicorn

app = FastAPI()


# Serializer
class RequestWithSingleExample(BaseModel):
    item_id: int = Field()
    name: str = Field()

    class Config:
        schema_extra = {
            'example': {
                'item_id': 1,
                'name': 'Name1'
            }
        }


# Serializer
class RequestWithMultiExamples(BaseModel):
    item_id: int = Field()
    name: str = Field()

    class Config:
        schema_extra = {
            'examples': {
                'example1': {
                    'summary': 'summary example1',
                    'description': 'descr example1',
                    'value': {
                        'item_id': 1,
                        'name': 'Name1'
                    }
                },
                'example2': {
                    'summary': 'summary example2',
                    'description': 'descr example2',
                    'value': {
                        'item_id': 2,
                        'name': 'Name2'
                    }
                }
            }
        }


# Depend
class RequestModelFactorySingleton(object):
    _models = {}

    def __new__(cls, model: BaseModel):
        if not cls._models.get(id(model)):
            cls._models[id(model)] = cls.query_processing_factory(model)
        return cls._models.get(id(model))

    @classmethod
    def query_processing_factory(cls, model: BaseModel):
        def processing_function(request: Request, items: model):
            # Some logic here
            return items

        return processing_function


# Endpoint with Depends and single example
# It is not possible to add multiple examples
@app.router.get('/single_example')
async def single_example(request: Request,
                         items: RequestWithSingleExample = Depends(
                             RequestModelFactorySingleton(RequestWithSingleExample))):
    return items


# Endpoint with multi examples
# It is not possible to use Depend
@app.router.get('/multi_examples')
async def multi_examples(request: Request,
                         items: RequestWithMultiExamples = Body(
                             examples=RequestWithMultiExamples.Config.schema_extra['examples'])):
    return items


# Endpoint with Depends and multi examples (examples are not displayed)
# Need support schema_extra['examples'] like schema_extra['example'] for working Depends with multi examples
@app.router.get('/multi_examples_with_depend')
async def multi_examples_with_depend(request: Request,
                                     items: RequestWithMultiExamples = Depends(
                                         RequestModelFactorySingleton(RequestWithMultiExamples))):
    return items


if __name__ == '__main__':
    uvicorn.run(app, host='0.0.0.0', port=8000)
```

### Description

Run the project from the example and open http://127.0.0.1:8000/redoc

* In '/single_example' endpoint uses Depends in body, but not possible to add multiple request examples
* In '/multi_examples' endpoint uses multiple request examples, but not possible to use Depends in body
* In '/multi_examples_with_depend' endpoint uses Depends and multiple request examples, but examples are not displayed

Need support Config.schema_extra['examples'] like a Condig.schema_extra['example'] for working 'Depends' with multi examples or some kind of workaround

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

3.8.10

**Comment by DavidBord** [comment]
@w11114m to customize OpenAPI examples, all is needed is to use a class that inherits from pydantic BaseModel, as specified here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/
Dependencies are not needed for customizing OpenAPI examples. Dependencies should be used for dependency injection, like for DB connections, as specified here: https://fastapi.tiangolo.com/tutorial/dependencies/
What are you trying to accomplish with RequestModelFactorySingleton? is it custom types? if so, I would recommend implementing it with pydantic. There are good examples for it here: https://docs.pydantic.dev/usage/types/#custom-data-types

**Comment by w11114m** [comment]
@DavidBord Thanks for the answer!
The fact is that I need to use Depends with a custom class to convert the request parameters into the request body. This needs to be done in Depends to support further pydantic validation of the resulting body.
I understand that this is an atypical implementation, but it is necessary.

For a better understanding, I will leave below a complete example of the RequestModelFactorySingleton class:

```
class RequestModelFactorySingleton(object):
    _models = {}

    def __new__(cls, model: BaseModel):
        if not cls._models.get(id(model)):
            cls._models[id(model)] = cls.query_processing_factory(model)
        return cls._models.get(id(model))

    @classmethod
    def query_processing_factory(cls, model: BaseModel):
        def processing_function(request: Request, items: model):
            if dict(request.query_params):
                try:
                    serialized_request = {
                        key: cls.transform_to_pyobj_or_return(value) for key, value in dict(request.query_params).items()}
                except json.JSONDecodeError as e:
                    raise ValidException(status.HTTP_400_BAD_REQUEST, f'Wrong json: {e}')
                items = items.__class__(**serialized_request)
            return items
        return processing_function

    @staticmethod
    def transform_to_pyobj_or_return(value):
        try:
            value = json.loads(value)
        except json.JSONDecodeError:
            pass
        return value
```

In this implementation, if you pass a pydantic class with an example request to schema_extra["example"] in RequestModelFactorySingleton, then the example will be displayed in the documentation, but multiple examples cannot be used. Because they have a completely different declaration that cannot be used with Depends:
https://fastapi.tiangolo.com/tutorial/schema-extra-example/#body-with-multiple-examples
